chat-widget/src/components/livechatwidget/common/endChat.ts

```diff
@@ -13,28 +13,53 @@ import { ILiveChatWidgetContext } from "../../../contexts/common/ILiveChatWidget
 import { PostChatSurveyMode } from "../../postchatsurveypanestateful/enums/PostChatSurveyMode";
 import { Constants } from "../../../common/Constants";
 import { ICustomEvent } from "@microsoft/omnichannel-chat-components/lib/types/interfaces/ICustomEvent";
-import { getWidgetEndChatEventName } from "../../../common/utils";
+import { addDelayInMs, getWidgetEndChatEventName } from "../../../common/utils";
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 const prepareEndChat = async (props: ILiveChatWidgetProps, chatSDK: any, setAdapter: any, setWebChatStyles: any, dispatch: Dispatch<ILiveChatWidgetAction>, adapter: any, state: ILiveChatWidgetContext) => {
     const isPostChatEnabled = state.domainStates.liveChatConfig?.LiveWSAndLiveChatEngJoin?.msdyn_postconversationsurveyenable;
     const postChatSurveyMode = state.domainStates.liveChatConfig?.LiveWSAndLiveChatEngJoin?.msdyn_postconversationsurveymode;
-    const conversationDetails = await chatSDK.getConversationDetails();
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    let conversationDetails: any = undefined;
+    try {
+        conversationDetails = await chatSDK.getConversationDetails();
+    }
+    catch (erorr) {
+        TelemetryHelper.logActionEvent(LogLevel.ERROR, {
+            Event: TelemetryEvent.GetConversationDetailsException,
+            ExceptionDetails: {
+                exception: `Failed to get conversation details: ${erorr}`
+            }
+        });
+    }
+
+    // Duplicate post chat loading on closing chat after PCS loaded
+    const isPostChatAlreadyLoaded = state.appStates.shouldShowPostChat;
+    if (isPostChatAlreadyLoaded === true) {
+        return;
+    }
+
     if (isPostChatEnabled === "true" && conversationDetails?.canRenderPostChat === Constants.truePascal) {
         const skipEndChatSDK = false;
         const skipCloseChat = true;
         await endChat(props, chatSDK, setAdapter, setWebChatStyles, dispatch, adapter, skipEndChatSDK, skipCloseChat, true);
+
         if (postChatSurveyMode === PostChatSurveyMode.Embed) {
+            dispatch({ type: LiveChatWidgetActionType.SET_POSTCHAT_LOADING, payload: true });
+            dispatch({ type: LiveChatWidgetActionType.SET_CONVERSATION_STATE, payload: ConversationState.Loading });
+            await addDelayInMs(Constants.PostChatLoadingDurationInMs);
```

Comment:
Why are we hardcoding this? We already got customers complaining the wait for post chat is too long

Reply:
As discussed reduced the time to 2secs